workflow/controller/operator_test.go

```diff
@@ -3778,37 +3778,6 @@ func TestNestedOptionalOutputArtifacts(t *testing.T) {
 	assert.Equal(t, wfv1.WorkflowSucceeded, woc.wf.Status.Phase)
 }
 
-var artifactGCOnWorkflowCompletion = `
-apiVersion: argoproj.io/v1alpha1
-kind: Workflow
-metadata:
-  generateName: artifact-passing-
-spec:
-  entrypoint: whalesay
-  artifactGC:
-    strategy: OnWorkflowCompletion
-  templates:
-  - name: whalesay
-    container:
-      image: docker/whalesay:latest
-      command: [sh, -c]
-      args: ["sleep 1; cowsay hello world | tee /tmp/hello_world.txt"]
-`
-
-func TestArtifactGCOnWorkflowCompletion(t *testing.T) {
-	cancel, controller := newController()
-	defer cancel()
-	ctx := context.Background()
-
-	wf := wfv1.MustUnmarshalWorkflow(artifactGCOnWorkflowCompletion)
-	woc := newWorkflowOperationCtx(wf, controller)
-	woc.operate(ctx)
-
-	assert.Equal(t, wfv1.WorkflowRunning, woc.wf.Status.Phase)
-	assert.Equal(t, 1, len(woc.wf.GetFinalizers()))
-	assert.Equal(t, "workflows.argoproj.io/artifact-gc", woc.wf.GetFinalizers()[0])
-}
-
 //  TestPodSpecLogForFailedPods tests PodSpec logging configuration
```

Comment:
The logic got moved into workflow_types.go, so the unit test moved to workflow_types_test.go. The remaining logic in operator.go seemed too simple to be worthy of a unit test, so I removed it.